File: MoveZeros/MoveZeros.py
def moveZeroes(nums):
    left = 0  # Pointer to place the next non-zero

    for right in range(len(nums)):
        if nums[right] != 0:
            # Swap only if left and right are at different indices
            nums[left], nums[right] = nums[right], nums[left]
            left += 1

    print(nums)

nums = [0,1,0,3,12]
moveZeroes(nums)

File: MoveZeros/MoveZeros.java
package MoveZeros;

public class MoveZeros {
    public static void main(String[] args) {
        int[] nums = {0,1,0,3,12};
        solution(nums);
    }

    public static void solution(int[] nums) {
        int left = 0;  
        for (int right = 0; right < nums.length; right++) {
            if (nums[right] != 0) {
                
                int temp = nums[left];
                nums[left] = nums[right];
                nums[right] = temp;

                left++;
            }
        }
        for (int i = 0; i < nums.length ;i++){
            System.out.print(nums[i] + ",") ;
        }
    }
}
